Source/WebKit/UIProcess/API/APIUIClient.h

```diff
@@ -230,7 +230,7 @@ class UIClient {
 
     virtual void queryPermission(const WTF::String& permissionName, SecurityOrigin& origin, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler) { completionHandler({ }); }
 
-#if ENABLE(WEBXR) && PLATFORM(COCOA)
+#if ENABLE(WEBXR)
     virtual void requestPermissionOnXRSessionFeatures(WebKit::WebPageProxy&, const WebCore::SecurityOriginData&, PlatformXR::SessionMode, const PlatformXR::Device::FeatureList& granted, const PlatformXR::Device::FeatureList& /* consentRequired */, const PlatformXR::Device::FeatureList& /* consentOptional */, const PlatformXR::Device::FeatureList& /* requiredFeaturesRequested */, const PlatformXR::Device::FeatureList& /* optionalFeaturesRequested */, CompletionHandler<void(std::optional<PlatformXR::Device::FeatureList>&&)>&& completionHandler) { completionHandler(granted); }
```

Comment:
The existing framework we have in the public API for permission requests should be a good fit. It involves triggering the [WebKitWebView::permission-request](https://wpewebkit.org/reference/stable/wpe-webkit-2.0/signal.WebView.permission-request.html) signal and providing [WebKitPermissionRequest](https://wpewebkit.org/reference/stable/wpe-webkit-2.0/iface.PermissionRequest.html) instances. The base `WebKitPermissionRequest` can be extended for XR permissions, and the mechanism is able to work asynchronously.

Anyway, I think wiring the permissions requests would be better done in a separate patch later on, and keep the focus on get the basics working in this one&mdash;my goal here is only to point out what could be a good option on this regard.

Reply:
Agree, at this point the make focus is to make WebXR work, this does not block the progress and we can leave with accept-all policy for now.